ksqldb-engine/src/test/java/io/confluent/ksql/analyzer/PullQueryValidatorTest.java

```diff
@@ -51,13 +59,22 @@
   private Into into;
   @Mock
   private RefinementInfo refinementInfo;
+  @Mock
+  private KsqlConfig ksqlConfig;
+  @Mock
+  private SingleColumn singleColumn;
+  @Mock
+  private ColumnReferenceExp columnReferenceExp;
+  @Mock
+  private ColumnReferenceExp columnReferenceExp2;
 
   private QueryValidator validator;
 
   @Before
   public void setUp() {
     validator = new PullQueryValidator();
-    when(analysis.getRefinementInfo()).thenReturn(Optional.of(RefinementInfo.of(OutputRefinement.FINAL)));
```

Comment:
Why is this being changed?

Reply:
Otherwise the test fails with `Pull queries don't support EMIT clauses` before the condition can reach the check I added. Previously this wasn't a problem as `PullQueryValidatorTest` only checks for failing conditions, and `!analysis.getRefinementInfo().isPresent()` was the last check.

Reply:
OK. IIUC you're saying that the fact that this line was present was a bug since the presence of this line causes all queries to fail validation. The bug was previously hidden since all the other expected failures happened before this one, but that's no longer true. Removing it (as you've done) sounds good.

Reply:
We seem to be on the same page- I'll keep it removed then.